File: solution/c#/Day13/Day13.Tests/ArticleBuilder.cs
namespace Day13.Tests
{
    public class ArticleBuilder
    {
        public const string Author = "Pablo Escobar";
        public const string CommentText = "Amazing article !!!";

        private readonly Dictionary<string, string> _comments = new();
        private readonly Bogus.Randomizer _random = new();

        public static ArticleBuilder AnArticle() => new();

        public ArticleBuilder Commented()
        {
            _comments.Add(CommentText, Author);
            return this;
        }

        public Article Build()
        {
            var article = new Article(_random.String(), _random.String());

            _comments
                .ToList()
                .ForEach(comment => article.AddComment(comment.Key, comment.Value));

            return article;
        }
    }
}

File: exercise/ts/day16/tests/articleBuilder.ts
import {Article} from "../src/blog";
import {faker} from "@faker-js/faker";

export const Author = "Pablo Escobar";
export const CommentText = "Amazing article !!!";

export class ArticleBuilder {
    private comments: Record<string, string> = {};

    static anArticle(): ArticleBuilder {
        return new ArticleBuilder();
    }

    commented(): ArticleBuilder {
        this.comments[CommentText] = Author;
        return this;
    }

    build(): Article {
        const article = new Article(faker.lorem.sentence(), faker.lorem.paragraph());

        Object.entries(this.comments).forEach(([text, author]) => {
            article.addComment(text, author);
        });

        return article;
    }
}